contracts/base/GuardManager.sol

```diff
@@ -20,13 +32,25 @@ interface Guard is IERC165 {
         address msgSender
     ) external;
 
-    function checkAfterExecution(bytes32 txHash, bool success) external;
+    /// @dev Checks if a module transaction should be allowed.
+    /// @param to The address of the recipient.
+    /// @param value The value of the transaction.
+    /// @param data The data of the transaction.
+    /// @param operation The operation type.
+    /// @param module The module that triggered the transaction.
+    function checkModuleTransaction(address to, uint256 value, bytes memory data, Enum.Operation operation, address module) external;
```

Comment:
Should we let this method return the `hash` that should be passed to the checkAfterExecution?

Reply:
I like the idea